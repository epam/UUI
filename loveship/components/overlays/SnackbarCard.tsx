import * as React from 'react';
import cx from 'classnames';
import { IHasChildren, IHasRawProps, INotification } from '@epam/uui-core';
import { IconContainer } from '@epam/uui-components';
import { IconButton } from '../buttons';
import { ReactComponent as CrossIcon } from './../icons/snackbar/cross.svg';
import { ReactComponent as InfoIcon } from './../icons/snackbar/info.svg';
import { ReactComponent as WarningIcon } from './../icons/snackbar/warning.svg';
import { ReactComponent as SuccessIcon } from './../icons/snackbar/success.svg';
import css from './SnackbarCard.scss';

export interface SnackbarCardProps extends INotification, IHasChildren, IHasRawProps<React.HTMLAttributes<HTMLDivElement>> {
    snackType: 'success' | 'warning' | 'info' | 'danger';
}

const SnackbarIcon = {
    success: SuccessIcon,
    warning: WarningIcon,
    danger: CrossIcon,
    info: InfoIcon,
};

export const SnackbarCard = React.forwardRef<HTMLDivElement, SnackbarCardProps>((props, ref) => (
    <div ref={ref} className={css.snackbar} {...props.rawProps}>
        <div className={cx([css[props.snackType], css.iconContainer])}>
            <IconContainer icon={SnackbarIcon[props.snackType]} cx={css.actionIcon} />
        </div>
        <div className={css.mainPath}>
            <div className={css.content}>{props.children}</div>
            <IconButton icon={CrossIcon} color="night600" onClick={props.onClose} cx={css.closeIcon} />
        </div>
    </div>
));
