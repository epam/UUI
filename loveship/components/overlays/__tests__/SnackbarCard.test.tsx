import React from 'react';
import renderer from 'react-test-renderer';
import { SnackbarCard } from '../SnackbarCard';

describe('SnackbarCard', () => {
    it('should be rendered correctly', () => {
        const tree = renderer.create(<SnackbarCard id={1} key="1" snackType="danger" onClose={jest.fn()} onSuccess={jest.fn()} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
