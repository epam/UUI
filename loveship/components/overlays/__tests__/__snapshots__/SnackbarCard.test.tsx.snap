// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnackbarCard should be rendered correctly 1`] = `
<div
  className="snackbar"
>
  <div
    className="danger iconContainer"
  >
    <div
      className="container uui-icon uui-enabled actionIcon"
      style={Object {}}
    >
      <svg
        className=""
      />
    </div>
  </div>
  <div
    className="mainPath"
  >
    <div
      className="content"
    />
    <button
      className="container uui-button-box uui-enabled -clickable root color-night600 closeIcon"
      href={null}
      onClick={[Function]}
      role="button"
      tabIndex={0}
      type="button"
    >
      <div
        className="container uui-icon uui-enabled"
        style={Object {}}
      >
        <svg
          className=""
        />
      </div>
    </button>
  </div>
</div>
`;
